### PR TITLE
Task 10

### PR DESCRIPTION
**Pull request by EDovgalo**
4) Not done

### CODE REVIEW

File: webpack.common.js [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-    extensions: ['.js', '.jsx', '.json', '.ts', '.tsx'],
-  },
-
-  plugins: [
-    new HtmlWebpackPlugin({
-      template: `${__dirname}/src/index.html`,
-    }),
-    new webpack.ProvidePlugin({
-      React: 'react',
-    }),
```
Comment (sulemanof) [inline webpack.common.js:25]: Why don't you save this plugin? This makes you import react into each file now and increase the diff. Was it necessary?

File: src/app/App.tsx [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
 import { AppFooter } from '../components/shared/AppFooter';
-import { HomePage } from '../components/pages/home/HomePage';
+import HomePage from '../components/pages/home/HomePage';
 import MovieDetailsPage from '../components/pages/details/MovieDetailsPage';
 import { SearchPage } from '../components/pages/search/SearchPage';
 import { NotFound } from '../components/pages/notFound/NotFound';
 import MovieSection from '../components/shared/movieSection/MovieSection';
 
-const App = (): JSX.Element => {
+const App = ({ Router, location, context, store }: any): JSX.Element => {
```
Comment (sulemanof) [inline src/app/App.tsx:15]: I am not sure why you've decided to pass `Router` as a prop here..
As far as I know, you could simply stay the `BrowserRouter` on the client side here, but only wrap with the static one on the server side.

File: src/helpers/utils.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import {object} from "prop-types";
```
Comment (sulemanof) [inline src/helpers/utils.ts:1]: Was this added unintentionally? 
And the second question, why don't you use eslint? I see a lot of error in your files..

Comment (EDovgalo) [reply src/helpers/utils.ts:1]: Fixed

File: src/app/App.tsx (hunk 2 above)
Comment (EDovgalo) [reply src/app/App.tsx:15]: Fixed

File: webpack.common.js (hunk 1 above)
Comment (EDovgalo) [reply webpack.common.js:25]: For tests. If not imported, into test files i got error:  ReferenceError: React is not defined

**Review by sulemanof — APPROVED** [review]

**Pull request merged**